diff --git a/Kptfile b/Kptfile
index 17a7822..98fa855 100644
--- a/Kptfile
+++ b/Kptfile
@@ -12,3 +12,22 @@ pipeline:
     - image: ghcr.io/kptdev/krm-functions-catalog/wasm/set-labels:v0.2.4
       configMap:
         tier: backend
+status:
+  conditions:
+    - type: Rendered
+      status: "True"
+      reason: RenderSuccess
+  renderStatus:
+    mutationSteps:
+      - image: ghcr.io/kptdev/krm-functions-catalog/wasm/set-namespace:v0.5.1
+        exitCode: 0
+        results:
+          - message: namespace [default] updated to "staging", 1 value(s) changed
+            severity: info
+          - message: all `depends-on` annotations are up-to-date. no `namespace` changed
+            severity: info
+      - image: ghcr.io/kptdev/krm-functions-catalog/wasm/set-labels:v0.2.4
+        exitCode: 0
+        results:
+          - message: set 4 labels in total
+            severity: info
diff --git a/resources.yaml b/resources.yaml
index eed43d6..c1de2b0 100644
--- a/resources.yaml
+++ b/resources.yaml
@@ -15,12 +15,20 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: nginx-deployment
+  namespace: staging
+  labels:
+    tier: backend
 spec:
   replicas: 3
+  selector:
+    matchLabels:
+      tier: backend
 ---
 apiVersion: custom.io/v1
 kind: Custom
 metadata:
   name: custom
+  labels:
+    tier: backend
 spec:
   image: nginx:1.2.3
