diff --git a/Kptfile b/Kptfile
index 3c63ab9..787b279 100644
--- a/Kptfile
+++ b/Kptfile
@@ -6,3 +6,8 @@ pipeline:
   mutators:
     - image: ghcr.io/kptdev/krm-functions-catalog/search-replace:latest
       configPath: search-replace-conf.yaml
+status:
+  conditions:
+    - type: Rendered
+      status: "True"
+      reason: RenderSuccess
diff --git a/resources.yaml b/resources.yaml
index f2eec52..114819d 100644
--- a/resources.yaml
+++ b/resources.yaml
@@ -16,7 +16,7 @@ kind: Deployment
 metadata:
   name: nginx-deployment
 spec:
-  replicas: 3
+  replicas: 4
 ---
 apiVersion: custom.io/v1
 kind: Custom
