diff --git a/Kptfile b/Kptfile
index 86dbe13..755cbd3 100644
--- a/Kptfile
+++ b/Kptfile
@@ -1,15 +1,43 @@
 apiVersion: kpt.dev/v1
 kind: Kptfile
 metadata:
+  name: bfs-subpkg-validator-fails
+  labels:
+    level: root
+    test: subpkg-fail
   annotations:
     kpt.dev/bfs-rendering: "true"
     kpt.dev/save-on-render-failure: "true"
-  name: bfs-subpkg-validator-fails
 info:
   description: BFS - Subpackage validator fails
 pipeline:
   mutators:
     - image: ghcr.io/kptdev/krm-functions-catalog/set-labels:v0.1.5
       configMap:
-        test: "subpkg-fail"
-        level: "root"
+        level: root
+        test: subpkg-fail
+status:
+  conditions:
+    - type: Rendered
+      status: "False"
+      reason: RenderFailed
+      message: |-
+        pkg.render: pkg .:
+        	pipeline.run: pkg ./subpkg: already handled error
+  renderStatus:
+    mutationSteps:
+      - image: ghcr.io/kptdev/krm-functions-catalog/set-labels:v0.1.5
+        exitCode: 0
+      - image: ghcr.io/kptdev/krm-functions-catalog/set-namespace:v0.2.0
+        exitCode: 0
+    validationSteps:
+      - image: ghcr.io/kptdev/krm-functions-catalog/starlark:latest
+        stderr: 'failed to evaluate function: error: function failure'
+        exitCode: 1
+        results:
+          - message: 'fail: intentional failure for testing'
+            severity: error
+        errorResults:
+          - message: 'fail: intentional failure for testing'
+            severity: error
+    errorSummary: 'ghcr.io/kptdev/krm-functions-catalog/starlark:latest: exit code 1'
diff --git a/deployment.yaml b/deployment.yaml
index 7123634..4db6211 100644
--- a/deployment.yaml
+++ b/deployment.yaml
@@ -3,5 +3,17 @@ kind: Deployment
 metadata:
   name: root-deployment
   namespace: default
+  labels:
+    level: root
+    test: subpkg-fail
 spec:
   replicas: 1
+  selector:
+    matchLabels:
+      level: root
+      test: subpkg-fail
+  template:
+    metadata:
+      labels:
+        level: root
+        test: subpkg-fail
diff --git a/subpkg/Kptfile b/subpkg/Kptfile
index 271c9a5..445ddf4 100644
--- a/subpkg/Kptfile
+++ b/subpkg/Kptfile
@@ -2,6 +2,10 @@ apiVersion: kpt.dev/v1
 kind: Kptfile
 metadata:
   name: subpkg
+  labels:
+    level: root
+    test: subpkg-fail
+  namespace: subpkg-ns
 info:
   description: Subpackage with failing validator
 pipeline:
diff --git a/subpkg/deployment.yaml b/subpkg/deployment.yaml
index 302322e..dfe3056 100644
--- a/subpkg/deployment.yaml
+++ b/subpkg/deployment.yaml
@@ -2,5 +2,18 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: sub-deployment
+  labels:
+    level: root
+    test: subpkg-fail
+  namespace: subpkg-ns
 spec:
   replicas: 2
+  selector:
+    matchLabels:
+      level: root
+      test: subpkg-fail
+  template:
+    metadata:
+      labels:
+        level: root
+        test: subpkg-fail
diff --git a/subpkg/starlark-fn-failure.yaml b/subpkg/starlark-fn-failure.yaml
index 55ceaae..4d01f35 100644
--- a/subpkg/starlark-fn-failure.yaml
+++ b/subpkg/starlark-fn-failure.yaml
@@ -2,6 +2,10 @@ apiVersion: fn.kpt.dev/v1alpha1
 kind: StarlarkRun
 metadata:
   name: always-fail
+  labels:
+    level: root
+    test: subpkg-fail
+  namespace: subpkg-ns
 source: |
   def fail_fn(items):
     fail("intentional failure for testing")
