diff --git a/Kptfile b/Kptfile
index 86dbe13..cba0f58 100644
--- a/Kptfile
+++ b/Kptfile
@@ -5,6 +5,9 @@ metadata:
     kpt.dev/bfs-rendering: "true"
     kpt.dev/save-on-render-failure: "true"
   name: bfs-subpkg-validator-fails
+  labels:
+    level: root
+    test: subpkg-fail
 info:
   description: BFS - Subpackage validator fails
 pipeline:
diff --git a/deployment.yaml b/deployment.yaml
index 7123634..594061d 100644
--- a/deployment.yaml
+++ b/deployment.yaml
@@ -3,5 +3,8 @@ kind: Deployment
 metadata:
   name: root-deployment
   namespace: default
+  labels:
+    level: root
+    test: subpkg-fail
 spec:
   replicas: 1
diff --git a/subpkg/Kptfile b/subpkg/Kptfile
index 7dd3a25..edd5b30 100644
--- a/subpkg/Kptfile
+++ b/subpkg/Kptfile
@@ -2,6 +2,9 @@ apiVersion: kpt.dev/v1
 kind: Kptfile
 metadata:
   name: subpkg
+  labels:
+    level: root
+    test: subpkg-fail
 info:
   description: Subpackage with failing validator
 pipeline:
diff --git a/subpkg/deployment.yaml b/subpkg/deployment.yaml
index 302322e..938e4c9 100644
--- a/subpkg/deployment.yaml
+++ b/subpkg/deployment.yaml
@@ -2,5 +2,9 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: sub-deployment
+  labels:
+    level: root
+    test: subpkg-fail
+  namespace: subpkg-ns
 spec:
   replicas: 2
