diff --git a/Kptfile b/Kptfile
index 93ae454..9caee20 100644
--- a/Kptfile
+++ b/Kptfile
@@ -2,6 +2,8 @@ apiVersion: kpt.dev/v1
 kind: Kptfile
 metadata:
   name: app
+  labels:
+    tier: backend
 pipeline:
   mutators:
     - image: ghcr.io/kptdev/krm-functions-catalog/set-namespace:latest
diff --git a/db/Kptfile b/db/Kptfile
index 79b7a5a..c1f496c 100644
--- a/db/Kptfile
+++ b/db/Kptfile
@@ -2,3 +2,5 @@ apiVersion: kpt.dev/v1
 kind: Kptfile
 metadata:
   name: db
+  labels:
+    tier: backend
diff --git a/db/resources.yaml b/db/resources.yaml
index f2eec52..e7f0c0b 100644
--- a/db/resources.yaml
+++ b/db/resources.yaml
@@ -15,6 +15,9 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: nginx-deployment
+  namespace: staging
+  labels:
+    tier: backend
 spec:
   replicas: 3
 ---
@@ -22,5 +25,7 @@ apiVersion: custom.io/v1
 kind: Custom
 metadata:
   name: custom
+  labels:
+    tier: backend
 spec:
   image: nginx:1.2.3
