diff --git a/Kptfile b/Kptfile
index f2fdc9a..c6b117b 100644
--- a/Kptfile
+++ b/Kptfile
@@ -8,3 +8,27 @@ pipeline:
       configMap:
         namespace: staging
     - image: ghcr.io/kptdev/krm-functions-catalog/set-labels:latest
+status:
+  conditions:
+    - type: Rendered
+      status: "False"
+      reason: RenderFailed
+      message: |-
+        pkg.render: pkg .:
+        	pipeline.run: already handled error
+  renderStatus:
+    mutationSteps:
+      - image: ghcr.io/kptdev/krm-functions-catalog/set-namespace:latest
+        exitCode: 0
+        results:
+          - message: namespace [default] updated to "staging", 1 value(s) changed
+            severity: info
+          - message: all `depends-on` annotations are up-to-date. no `namespace` changed
+            severity: info
+      - image: ghcr.io/kptdev/krm-functions-catalog/set-labels:latest
+        stderr: 'failed to evaluate function: error: function failure'
+        exitCode: 1
+        results:
+          - message: '`FunctionConfig` is not given'
+            severity: info
+    errorSummary: 'ghcr.io/kptdev/krm-functions-catalog/set-labels:latest: exit code 1'
