diff --git a/Kptfile b/Kptfile
index 2c6e965..481dc83 100644
--- a/Kptfile
+++ b/Kptfile
@@ -7,3 +7,59 @@ pipeline:
     - image: ghcr.io/kptdev/krm-functions-catalog/kubeconform:latest
       configMap:
         strict: "true"
+status:
+  conditions:
+    - type: Rendered
+      status: "False"
+      reason: RenderFailed
+      message: |-
+        pkg.render: pkg .:
+        	pipeline.run: already handled error
+  renderStatus:
+    validationSteps:
+      - image: ghcr.io/kptdev/krm-functions-catalog/kubeconform:latest
+        stderr: 'failed to evaluate function: error: function failure'
+        exitCode: 1
+        results:
+          - message: missing properties 'selector', 'template'
+            severity: error
+            resourceRef:
+              apiVersion: apps/v1
+              kind: Deployment
+              name: nginx-deployment
+            field:
+              path: spec
+            file:
+              path: resources.yaml
+          - message: got string, want null or integer
+            severity: error
+            resourceRef:
+              apiVersion: apps/v1
+              kind: Deployment
+              name: nginx-deployment
+            field:
+              path: spec.replicas
+            file:
+              path: resources.yaml
+        errorResults:
+          - message: missing properties 'selector', 'template'
+            severity: error
+            resourceRef:
+              apiVersion: apps/v1
+              kind: Deployment
+              name: nginx-deployment
+            field:
+              path: spec
+            file:
+              path: resources.yaml
+          - message: got string, want null or integer
+            severity: error
+            resourceRef:
+              apiVersion: apps/v1
+              kind: Deployment
+              name: nginx-deployment
+            field:
+              path: spec.replicas
+            file:
+              path: resources.yaml
+    errorSummary: 'ghcr.io/kptdev/krm-functions-catalog/kubeconform:latest: exit code 1'
