diff --git a/Kptfile b/Kptfile
index d8eebcb..a84934e 100644
--- a/Kptfile
+++ b/Kptfile
@@ -5,3 +5,12 @@ metadata:
 pipeline:
   mutators:
     - image: ghcr.io/kptdev/krm-functions-catalog/drop-comments:latest
+status:
+  conditions:
+    - type: Rendered
+      status: "True"
+      reason: RenderSuccess
+  renderStatus:
+    mutationSteps:
+      - image: ghcr.io/kptdev/krm-functions-catalog/drop-comments:latest
+        exitCode: 0
