diff --git a/Kptfile b/Kptfile
index 5053b98..b29e7dd 100644
--- a/Kptfile
+++ b/Kptfile
@@ -5,5 +5,10 @@ metadata:
 pipeline:
   mutators:
     - image: ghcr.io/kptdev/krm-functions-catalog/set-labels:v0.1.5
       configMap:
         env: production
-      condition: "resources.exists(r, r.kind == 'ConfigMap' && r.metadata.name == 'app-config')"
+      condition: resources.exists(r, r.kind == 'ConfigMap' && r.metadata.name == 'app-config')
+status:
+  conditions:
+    - type: Rendered
+      status: "True"
+      reason: RenderSuccess
