diff --git a/Kptfile b/Kptfile
index 5d377d4..784c77a 100644
--- a/Kptfile
+++ b/Kptfile
@@ -4,5 +4,10 @@ metadata:
   name: app
 pipeline:
   mutators:
-  - image: ghcr.io/kptdev/krm-functions-catalog/starlark:latest
-    configPath: starlark-fn.yaml
+    - image: ghcr.io/kptdev/krm-functions-catalog/starlark:latest
+      configPath: starlark-fn.yaml
+status:
+  conditions:
+    - type: Rendered
+      status: "True"
+      reason: RenderSuccess
diff --git a/deployment.yaml b/newfilename.yaml
similarity index 100%
rename from deployment.yaml
rename to newfilename.yaml
diff --git a/starlark-fn.yaml b/starlark-fn.yaml
index 247c2ae..0ef906e 100644
--- a/starlark-fn.yaml
+++ b/starlark-fn.yaml
@@ -11,7 +11,6 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
 apiVersion: fn.kpt.dev/v1alpha1
 kind: StarlarkRun
 metadata:
