diff --git a/Kptfile b/Kptfile
index 26aa62c..5589489 100644
--- a/Kptfile
+++ b/Kptfile
@@ -2,6 +2,8 @@ apiVersion: kpt.dev/v1
 kind: Kptfile
 metadata:
   name: app-with-generator
+  labels:
+    tier: db
 pipeline:
   mutators:
     - image: ghcr.io/kptdev/krm-functions-catalog/set-namespace:latest
@@ -10,3 +12,36 @@ pipeline:
     - image: ghcr.io/kptdev/krm-functions-catalog/set-labels:latest
       configMap:
         tier: db
+status:
+  conditions:
+    - type: Rendered
+      status: "True"
+      reason: RenderSuccess
+  renderStatus:
+    mutationSteps:
+      - image: ghcr.io/kptdev/krm-functions-catalog/starlark:latest
+        exitCode: 0
+      - image: ghcr.io/kptdev/krm-functions-catalog/set-namespace:latest
+        exitCode: 0
+        results:
+          - message: namespace [default] updated to "db", 2 value(s) changed
+            severity: info
+          - message: all `depends-on` annotations are up-to-date. no `namespace` changed
+            severity: info
+      - image: ghcr.io/kptdev/krm-functions-catalog/set-labels:latest
+        exitCode: 0
+        results:
+          - message: set 7 labels in total
+            severity: info
+      - image: ghcr.io/kptdev/krm-functions-catalog/set-namespace:latest
+        exitCode: 0
+        results:
+          - message: namespace [db default] updated to "staging", 3 value(s) changed
+            severity: info
+          - message: all `depends-on` annotations are up-to-date. no `namespace` changed
+            severity: info
+      - image: ghcr.io/kptdev/krm-functions-catalog/set-labels:latest
+        exitCode: 0
+        results:
+          - message: set 11 labels in total
+            severity: info
diff --git a/db/Kptfile b/db/Kptfile
index 4538e5e..95ee6cd 100644
--- a/db/Kptfile
+++ b/db/Kptfile
@@ -2,6 +2,9 @@ apiVersion: kpt.dev/v1
 kind: Kptfile
 metadata:
   name: db
+  labels:
+    app: backend
+    tier: db
 pipeline:
   mutators:
     - image: ghcr.io/kptdev/krm-functions-catalog/starlark:latest
diff --git a/db/deployment_httpbin.yaml b/db/deployment_httpbin.yaml
new file mode 100644
index 0000000..ffdf484
--- /dev/null
+++ b/db/deployment_httpbin.yaml
@@ -0,0 +1,25 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: httpbin
+  namespace: staging
+  labels:
+    app: backend
+    tier: db
+spec:
+  replicas: 4
+  template:
+    spec:
+      containers:
+      - image: kennethreitz/httpbin
+        name: httpbin
+        ports:
+        - containerPort: 9876
+    metadata:
+      labels:
+        app: backend
+        tier: db
+  selector:
+    matchLabels:
+      app: backend
+      tier: db
diff --git a/db/resources.yaml b/db/resources.yaml
index dabe43c..25f77c1 100644
--- a/db/resources.yaml
+++ b/db/resources.yaml
@@ -15,5 +15,13 @@ apiVersion: apps/v1
 kind: StatefulSet
 metadata:
   name: db
+  namespace: staging
+  labels:
+    app: backend
+    tier: db
 spec:
   replicas: 3
+  selector:
+    matchLabels:
+      app: backend
+      tier: db
diff --git a/db/starlark-httpbin.yaml b/db/starlark-httpbin.yaml
index e52e48f..7aad15e 100644
--- a/db/starlark-httpbin.yaml
+++ b/db/starlark-httpbin.yaml
@@ -15,6 +15,9 @@ apiVersion: fn.kpt.dev/v1alpha1
 kind: StarlarkRun
 metadata:
   name: httpbin-gen
+  labels:
+    app: backend
+    tier: db
 source: |-
   httpbin_deployment = {
      "apiVersion": "apps/v1",
diff --git a/resources.yaml b/resources.yaml
index f2eec52..dcf72b1 100644
--- a/resources.yaml
+++ b/resources.yaml
@@ -15,12 +15,20 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: nginx-deployment
+  namespace: staging
+  labels:
+    tier: db
 spec:
   replicas: 3
+  selector:
+    matchLabels:
+      tier: db
 ---
 apiVersion: custom.io/v1
 kind: Custom
 metadata:
   name: custom
+  labels:
+    tier: db
 spec:
   image: nginx:1.2.3
