diff --git a/Kptfile b/Kptfile
index 1307fb5..cdd5362 100644
--- a/Kptfile
+++ b/Kptfile
@@ -10,3 +10,15 @@ pipeline:
     - image: ghcr.io/kptdev/krm-functions-catalog/set-labels:v0.1.5
       configMap:
         tier: backend
+status:
+  conditions:
+    - type: Rendered
+      status: "False"
+      reason: RenderFailed
+      message: |-
+        pkg.render: pkg .:
+        	pipeline.run: input resource list must contain only KRM resources: non-krm.yaml: resource must have `apiVersion`
+  renderStatus:
+    mutationSteps:
+      - image: ghcr.io/kptdev/krm-functions-catalog/set-namespace:v0.2.0
+        exitCode: 0
