diff --git a/Kptfile b/Kptfile
index 701e0a1..b7a3e2d 100644
--- a/Kptfile
+++ b/Kptfile
@@ -4,3 +4,17 @@ metadata:
   name: root-pkg
 info:
   description: sample description
+status:
+  conditions:
+    - type: Rendered
+      status: "True"
+      reason: RenderSuccess
+  renderStatus:
+    mutationSteps:
+      - image: ghcr.io/kptdev/krm-functions-catalog/set-namespace:latest
+        exitCode: 0
+        results:
+          - message: namespace [default] updated to "dev", 1 value(s) changed
+            severity: info
+          - message: all `depends-on` annotations are up-to-date. no `namespace` changed
+            severity: info
diff --git a/pkg-a/pkg-a/resources.yaml b/pkg-a/pkg-a/resources.yaml
index f2eec52..1934a8f 100644
--- a/pkg-a/pkg-a/resources.yaml
+++ b/pkg-a/pkg-a/resources.yaml
@@ -15,6 +15,7 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: nginx-deployment
+  namespace: dev
 spec:
   replicas: 3
 ---
