diff --git a/Kptfile b/Kptfile
index e60b7a4..064d8e6 100644
--- a/Kptfile
+++ b/Kptfile
@@ -7,6 +7,31 @@ pipeline:
     - image: ghcr.io/kptdev/krm-functions-catalog/set-namespace:latest
       configMap:
         namespace: staging
-    - image: ghcr.io/kptdev/krm-functions-catalog/dne # non-existing image
+    - image: ghcr.io/kptdev/krm-functions-catalog/dne
       configMap:
         tier: backend
+status:
+  conditions:
+    - type: Rendered
+      status: "False"
+      reason: RenderFailed
+      message: |-
+        pkg.render: pkg .:
+        	pipeline.run: already handled error
+  renderStatus:
+    mutationSteps:
+      - image: ghcr.io/kptdev/krm-functions-catalog/set-namespace:latest
+        exitCode: 0
+        results:
+          - message: namespace [default] updated to "staging", 1 value(s) changed
+            severity: info
+          - message: all `depends-on` annotations are up-to-date. no `namespace` changed
+            severity: info
+      - image: ghcr.io/kptdev/krm-functions-catalog/dne:latest
+        stderr: |-
+          docker: Error response from daemon: error from registry: denied
+          denied
+
+          Run 'docker run --help' for more information
+        exitCode: 125
+    errorSummary: 'ghcr.io/kptdev/krm-functions-catalog/dne:latest: exit code 125'
