diff --git a/Kptfile b/Kptfile
index 11012de..c9efd96 100644
--- a/Kptfile
+++ b/Kptfile
@@ -7,6 +7,14 @@ pipeline:
     - image: ghcr.io/kptdev/krm-functions-catalog/set-namespace:v0.2.0
       configMap:
         namespace: staging
-    - image: ghcr.io/kptdev/krm-functions-catalog/dne # non-existing image
+    - image: ghcr.io/kptdev/krm-functions-catalog/dne
       configMap:
         tier: backend
+status:
+  conditions:
+    - type: Rendered
+      status: "False"
+      reason: RenderFailed
+      message: |-
+        pkg.render: pkg .:
+        	pipeline.run: already handled error
