diff --git a/Kptfile b/Kptfile
index 7c3b09a..2b496da 100644
--- a/Kptfile
+++ b/Kptfile
@@ -1,14 +1,37 @@
 apiVersion: kpt.dev/v1
 kind: Kptfile
 metadata:
+  name: dfs-subpkg-validator-fails
   annotations:
     kpt.dev/save-on-render-failure: "true"
-  name: dfs-subpkg-validator-fails
 info:
   description: DFS - Subpackage validator fails
 pipeline:
   mutators:
     - image: ghcr.io/kptdev/krm-functions-catalog/set-labels:v0.1.5
       configMap:
-        test: "subpkg-fail"
-        level: "root"
+        level: root
+        test: subpkg-fail
+status:
+  conditions:
+    - type: Rendered
+      status: "False"
+      reason: RenderFailed
+      message: |-
+        pkg.render: pkg ./subpkg:
+        	pipeline.run: already handled error
+  renderStatus:
+    mutationSteps:
+      - image: ghcr.io/kptdev/krm-functions-catalog/set-namespace:v0.2.0
+        exitCode: 0
+    validationSteps:
+      - image: ghcr.io/kptdev/krm-functions-catalog/starlark:latest
+        stderr: 'failed to evaluate function: error: function failure'
+        exitCode: 1
+        results:
+          - message: 'fail: intentional failure for testing'
+            severity: error
+        errorResults:
+          - message: 'fail: intentional failure for testing'
+            severity: error
+    errorSummary: 'ghcr.io/kptdev/krm-functions-catalog/starlark:latest: exit code 1'
diff --git a/subpkg/Kptfile b/subpkg/Kptfile
index 271c9a5..cbd2fa4 100644
--- a/subpkg/Kptfile
+++ b/subpkg/Kptfile
@@ -2,6 +2,7 @@ apiVersion: kpt.dev/v1
 kind: Kptfile
 metadata:
   name: subpkg
+  namespace: subpkg-ns
 info:
   description: Subpackage with failing validator
 pipeline:
diff --git a/subpkg/deployment.yaml b/subpkg/deployment.yaml
index 302322e..98ff830 100644
--- a/subpkg/deployment.yaml
+++ b/subpkg/deployment.yaml
@@ -2,5 +2,6 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: sub-deployment
+  namespace: subpkg-ns
 spec:
   replicas: 2
diff --git a/subpkg/starlark-fn-failure.yaml b/subpkg/starlark-fn-failure.yaml
index 55ceaae..3a4003f 100644
--- a/subpkg/starlark-fn-failure.yaml
+++ b/subpkg/starlark-fn-failure.yaml
@@ -2,6 +2,7 @@ apiVersion: fn.kpt.dev/v1alpha1
 kind: StarlarkRun
 metadata:
   name: always-fail
+  namespace: subpkg-ns
 source: |
   def fail_fn(items):
     fail("intentional failure for testing")
