diff --git a/Kptfile b/Kptfile
index 17ef3d8..93810be 100644
--- a/Kptfile
+++ b/Kptfile
@@ -1,17 +1,42 @@
 apiVersion: kpt.dev/v1
 kind: Kptfile
 metadata:
+  name: bfs-parent-validator-fails
+  labels:
+    test: parent-fail
   annotations:
     kpt.dev/bfs-rendering: "true"
     kpt.dev/save-on-render-failure: "true"
-  name: bfs-parent-validator-fails
 info:
   description: BFS - Parent validator fails
 pipeline:
   mutators:
     - image: ghcr.io/kptdev/krm-functions-catalog/set-labels:v0.1.5
       configMap:
-        test: "parent-fail"
+        test: parent-fail
   validators:
     - image: ghcr.io/kptdev/krm-functions-catalog/starlark:latest
       configPath: starlark-fn-failure.yaml
+status:
+  conditions:
+    - type: Rendered
+      status: "False"
+      reason: RenderFailed
+      message: |-
+        pkg.render: pkg .:
+        	pipeline.run: already handled error
+  renderStatus:
+    mutationSteps:
+      - image: ghcr.io/kptdev/krm-functions-catalog/set-labels:v0.1.5
+        exitCode: 0
+    validationSteps:
+      - image: ghcr.io/kptdev/krm-functions-catalog/starlark:latest
+        stderr: 'failed to evaluate function: error: function failure'
+        exitCode: 1
+        results:
+          - message: 'fail: intentional failure for testing'
+            severity: error
+        errorResults:
+          - message: 'fail: intentional failure for testing'
+            severity: error
+    errorSummary: 'ghcr.io/kptdev/krm-functions-catalog/starlark:latest: exit code 1'
diff --git a/configmap.yaml b/configmap.yaml
index 20a54f6..ad595de 100644
--- a/configmap.yaml
+++ b/configmap.yaml
@@ -2,5 +2,7 @@ apiVersion: v1
 kind: ConfigMap
 metadata:
   name: root-config
+  labels:
+    test: parent-fail
 data:
   key: value
diff --git a/starlark-fn-failure.yaml b/starlark-fn-failure.yaml
index 55ceaae..eb4b975 100644
--- a/starlark-fn-failure.yaml
+++ b/starlark-fn-failure.yaml
@@ -2,6 +2,8 @@ apiVersion: fn.kpt.dev/v1alpha1
 kind: StarlarkRun
 metadata:
   name: always-fail
+  labels:
+    test: parent-fail
 source: |
   def fail_fn(items):
     fail("intentional failure for testing")
diff --git a/subpkg/Kptfile b/subpkg/Kptfile
index e940ae8..88a8d4e 100644
--- a/subpkg/Kptfile
+++ b/subpkg/Kptfile
@@ -2,6 +2,8 @@ apiVersion: kpt.dev/v1
 kind: Kptfile
 metadata:
   name: subpkg
+  labels:
+    test: parent-fail
 info:
   description: Subpackage that succeeds
 pipeline:
diff --git a/subpkg/service.yaml b/subpkg/service.yaml
index 276bf9e..fe08496 100644
--- a/subpkg/service.yaml
+++ b/subpkg/service.yaml
@@ -2,6 +2,10 @@ apiVersion: v1
 kind: Service
 metadata:
   name: sub-service
+  labels:
+    test: parent-fail
 spec:
   ports:
   - port: 80
+  selector:
+    test: parent-fail
