diff --git a/Kptfile b/Kptfile
index 1667ca3..1fec05d 100644
--- a/Kptfile
+++ b/Kptfile
@@ -4,11 +4,61 @@ metadata:
   name: no-save-on-render-failure
 pipeline:
   mutators:
-  - image: ghcr.io/kptdev/krm-functions-catalog/set-namespace:latest
-    configMap:
-      namespace: staging
-  - image: ghcr.io/kptdev/krm-functions-catalog/set-annotations:latest
-    configMap:
-      app: myapp
-  - image: ghcr.io/kptdev/krm-functions-catalog/starlark:latest
-    configPath: starlark-fn-failure.yaml
+    - image: ghcr.io/kptdev/krm-functions-catalog/set-namespace:latest
+      configMap:
+        namespace: staging
+    - image: ghcr.io/kptdev/krm-functions-catalog/set-annotations:latest
+      configMap:
+        app: myapp
+    - image: ghcr.io/kptdev/krm-functions-catalog/starlark:latest
+      configPath: starlark-fn-failure.yaml
+status:
+  conditions:
+    - type: Rendered
+      status: "False"
+      reason: RenderFailed
+      message: |-
+        pkg.render: pkg .:
+        	pipeline.run: already handled error
+  renderStatus:
+    mutationSteps:
+      - image: ghcr.io/kptdev/krm-functions-catalog/set-namespace:latest
+        exitCode: 0
+        results:
+          - message: namespace [default] updated to "staging", 1 value(s) changed
+            severity: info
+          - message: all `depends-on` annotations are up-to-date. no `namespace` changed
+            severity: info
+      - image: ghcr.io/kptdev/krm-functions-catalog/set-annotations:latest
+        exitCode: 0
+        results:
+          - message: 'set annotations: {"app":"myapp"}'
+            field:
+              path: metadata.annotations
+            file:
+              path: Kptfile
+          - message: 'set annotations: {"app":"myapp"}'
+            field:
+              path: metadata.annotations
+            file:
+              path: resources.yaml
+          - message: 'set annotations: {"app":"myapp"}'
+            field:
+              path: spec.template.metadata.annotations
+            file:
+              path: resources.yaml
+          - message: 'set annotations: {"app":"myapp"}'
+            field:
+              path: metadata.annotations
+            file:
+              path: starlark-fn-failure.yaml
+      - image: ghcr.io/kptdev/krm-functions-catalog/starlark:latest
+        stderr: 'failed to evaluate function: error: function failure'
+        exitCode: 1
+        results:
+          - message: 'fail: intentional failure for testing'
+            severity: error
+        errorResults:
+          - message: 'fail: intentional failure for testing'
+            severity: error
+    errorSummary: 'ghcr.io/kptdev/krm-functions-catalog/starlark:latest: exit code 1'
