diff --git a/Kptfile b/Kptfile
index 6e26cd3..fb2aa71 100644
--- a/Kptfile
+++ b/Kptfile
@@ -4,12 +4,20 @@ metadata:
   name: no-save-on-render-failure
 pipeline:
   mutators:
-  - image: ghcr.io/kptdev/krm-functions-catalog/set-namespace:v0.2.0
-    configMap:
-      namespace: staging
-  - image: ghcr.io/kptdev/krm-functions-catalog/set-annotations:v0.1.4
-    configMap:
-      app: myapp
-  - image: invalid-image:v0.0.0
-    configMap:
-      tier: backend
+    - image: ghcr.io/kptdev/krm-functions-catalog/set-namespace:v0.2.0
+      configMap:
+        namespace: staging
+    - image: ghcr.io/kptdev/krm-functions-catalog/set-annotations:v0.1.4
+      configMap:
+        app: myapp
+    - image: invalid-image:v0.0.0
+      configMap:
+        tier: backend
+status:
+  conditions:
+    - type: Rendered
+      status: "False"
+      reason: RenderFailed
+      message: |-
+        pkg.render: pkg .:
+        	pipeline.run: already handled error
