diff --git a/Kptfile b/Kptfile
index f764615..877725d 100644
--- a/Kptfile
+++ b/Kptfile
@@ -1,14 +1,16 @@
-apiVersion: kpt.dev/v1
-kind: Kptfile
-metadata:
-  annotations:
-    kpt.dev/bfs-rendering: "true"
-    kpt.dev/save-on-render-failure: "true"
-  name: bfs-subpkg-mutator-fails
-info:
-  description: BFS - Subpackage mutator fails
-pipeline:
-  mutators:
-    - image: ghcr.io/kptdev/krm-functions-catalog/set-labels:v0.1.5
-      configMap:
-        test: "mutator-fail"
+apiVersion: kpt.dev/v1
+kind: Kptfile
+metadata:
+  annotations:
+    kpt.dev/bfs-rendering: "true"
+    kpt.dev/save-on-render-failure: "true"
+  name: bfs-subpkg-mutator-fails
+  labels:
+    test: mutator-fail
+info:
+  description: BFS - Subpackage mutator fails
+pipeline:
+  mutators:
+    - image: ghcr.io/kptdev/krm-functions-catalog/set-labels:v0.1.5
+      configMap:
+        test: mutator-fail
diff --git a/deployment.yaml b/deployment.yaml
index d4db082..6ed4201 100644
--- a/deployment.yaml
+++ b/deployment.yaml
@@ -1,6 +1,15 @@
-apiVersion: apps/v1
-kind: Deployment
-metadata:
-  name: root-deployment
-spec:
-  replicas: 1
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: root-deployment
+  labels:
+    test: mutator-fail
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      test: mutator-fail
+  template:
+    metadata:
+      labels:
+        test: mutator-fail
diff --git a/subpkg/Kptfile b/subpkg/Kptfile
index ae05dfe..19e913c 100644
--- a/subpkg/Kptfile
+++ b/subpkg/Kptfile
@@ -1,12 +1,14 @@
-apiVersion: kpt.dev/v1
-kind: Kptfile
-metadata:
-  name: subpkg
-info:
-  description: Subpackage with failing mutator
-pipeline:
-  mutators:
-    - image: ghcr.io/kptdev/krm-functions-catalog/set-namespace:latest
-      configMap:
-        namespace: subpkg-ns
-    - image: ghcr.io/kptdev/krm-functions-catalog/invalid-image:v0.0.0
+apiVersion: kpt.dev/v1
+kind: Kptfile
+metadata:
+  name: subpkg
+  labels:
+    test: mutator-fail
+info:
+  description: Subpackage with failing mutator
+pipeline:
+  mutators:
+    - image: ghcr.io/kptdev/krm-functions-catalog/set-namespace:latest
+      configMap:
+        namespace: subpkg-ns
+    - image: ghcr.io/kptdev/krm-functions-catalog/invalid-image:v0.0.0
diff --git a/subpkg/deployment.yaml b/subpkg/deployment.yaml
index 1c4bc07..546fe5c 100644
--- a/subpkg/deployment.yaml
+++ b/subpkg/deployment.yaml
@@ -1,6 +1,16 @@
-apiVersion: apps/v1
-kind: Deployment
-metadata:
-  name: sub-deployment
-spec:
-  replicas: 2
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: sub-deployment
+  labels:
+    test: mutator-fail
+  namespace: subpkg-ns
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      test: mutator-fail
+  template:
+    metadata:
+      labels:
+        test: mutator-fail
