diff --git a/Kptfile b/Kptfile
index 2210a5f..9383e82 100755
--- a/Kptfile
+++ b/Kptfile
@@ -1,8 +1,19 @@
-apiVersion: kpt.dev/v1
-kind: Kptfile
-metadata:
-  name: app
-pipeline:
-  mutators:
-    - image: ghcr.io/kptdev/krm-functions-catalog/no-op
-      condition: "resources.exists(r, r.kind == 'ConfigMap' && r.metadata.name == 'app-config')"
+apiVersion: kpt.dev/v1
+kind: Kptfile
+metadata:
+  name: app
+pipeline:
+  mutators:
+    - image: ghcr.io/kptdev/krm-functions-catalog/no-op
+status:
+  conditions:
+    - type: Rendered
+      status: "True"
+      reason: RenderSuccess
+  renderStatus:
+    mutationSteps:
+      - image: ghcr.io/kptdev/krm-functions-catalog/no-op:latest
+        exitCode: 0
+        skipped: true
diff --git a/resources.yaml b/resources.yaml
index d9eb99a..c8cdecf 100755
--- a/resources.yaml
+++ b/resources.yaml
@@ -1,6 +1,6 @@
-apiVersion: apps/v1
-kind: Deployment
-metadata:
-  name: my-app
-spec:
-  replicas: 1
\ No newline at end of file
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-app
+spec:
+  replicas: 1
