diff --git a/Kptfile b/Kptfile
index ef99dad..2d02fa0 100644
--- a/Kptfile
+++ b/Kptfile
@@ -6,3 +6,17 @@ pipeline:
   mutators:
     - image: ghcr.io/kptdev/krm-functions-catalog/starlark:latest
       configPath: starlark-gen-duplicate-path.yaml
+status:
+  conditions:
+    - type: Rendered
+      status: "False"
+      reason: RenderFailed
+      message: |-
+        pkg.render: pkg .:
+        	pipeline.run: resource at path "resources.yaml" and index "0" already exists
+  renderStatus:
+    mutationSteps:
+      - image: ghcr.io/kptdev/krm-functions-catalog/starlark:latest
+        executionError: resource at path "resources.yaml" and index "0" already exists
+        exitCode: 1
+    errorSummary: 'ghcr.io/kptdev/krm-functions-catalog/starlark:latest: resource at path "resources.yaml" and index "0" already exists'
