diff --git a/Kptfile b/Kptfile
index 1307fb5..4433dbf 100644
--- a/Kptfile
+++ b/Kptfile
@@ -10,3 +10,13 @@ pipeline:
     - image: ghcr.io/kptdev/krm-functions-catalog/set-labels:v0.1.5
       configMap:
         tier: backend
+status:
+  conditions:
+    - type: Rendered
+      status: "False"
+      observedgeneration: 0
+      lasttransitiontime: "0001-01-01T00:00:00Z"
+      reason: RenderFailed
+      message: |-
+        pkg.render: pkg .:
+        	pkg.Subpackages: pkg ./db: error reading Kptfile at "./db": yaml: line 10: mapping values are not allowed in this context
